src/security.ts

```diff
@@ -12,10 +12,9 @@ export interface InputSecurity<K extends string> {
   name: K;
 }
 
-/** @todo add constraints similar to InputSecurity, rename to just HeaderSecurity */
-export interface CustomHeaderSecurity {
+export interface HeaderSecurity<K extends string> {
   type: "header";
-  name: string;
+  name: K;
```

Comment:
or... maybe not.
because it would require to opt-in for `headers` as input source, while there is still an opportunity to access `request.headers` in middleware directly.